res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1394,27 +1418,31 @@ PioneerDDJSX.tempoResetButton = function(channel, control, value, status, group)
 PioneerDDJSX.autoLoopButton = function(channel, control, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
-            engine.setValue(group, "reloop_exit", true);
+            engine.setValue(group, "reloop_toggle", true);
         } else {
-            engine.setValue(group, "beatloop_" + 4 + "_toggle", true);
+            engine.setValue(group, "beatloop_activate", true);
         }
     }
 };
 
 PioneerDDJSX.loopActiveButton = function(channel, control, value, status, group) {
-    script.toggleControl(group, "reloop_exit");
+    script.toggleControl(group, "reloop_toggle");
```

Comment:
As I realized with #1271, toggling Controls on each button down/up event can cause strange situations if the Control does not start in the expected state. For example, if reloop_toggle happens to be 1 for whatever reason when the button is pressed, it will be set to 0 on button down and reset to 1 on button up, which will not produce the expected behavior. It's better to set the Control to 1 on button press and 0 on button release, regardless of the state of the Control before the button was pressed.